src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1548,7 +1548,19 @@ private static void ComputeDeclarationsInNode(SemanticModel semanticModel, ISymb
                             if (shouldExecuteOperationActions || shouldExecuteOperationBlockActions)
                             {
                                 var operationBlocksToAnalyze = GetOperationBlocksToAnalyze(executableCodeBlocks, semanticModel, cancellationToken);
-                                var operationsToAnalyze = GetOperationsToAnalyze(operationBlocksToAnalyze);
+                                var operationsToAnalyze = ImmutableArray<IOperation>.Empty;
+                                try
+                                {
+                                    // OperationWalker can throw `InsufficientExecutionStackException`
+                                    operationsToAnalyze = GetOperationsToAnalyze(operationBlocksToAnalyze);
+                                }
+                                catch (Exception ex) when (FatalError.ReportWithoutCrashUnlessCanceled(ex))
```

Comment:
Do we really want to create a Watson for an InsufficientExecutionStackException?